server/src/main/java/org/opensearch/wlm/QueryGroupTask.java

```diff
@@ -92,6 +94,10 @@ public long getElapsedTime() {
         return nanoTimeSupplier.getAsLong() - getStartTimeNanos();
     }
 
+    public boolean isEligibleForTracking() {
```

Comment:
On the surface this API doesn't make sense to me. What is the purpose of creating an instance that is ineligible for tracking?

Reply:
@andrross this is to basically filter out the tasks are created in `TransportAction` class and have still not reached the corresponding specialised transport classes e,g; `TransportSearchAction`. These specialised transport classes are responsible for setting the `queryGroupId`.

Reply:
Probably we should rename this for better clarity. How about `isQueryGroupSet`?